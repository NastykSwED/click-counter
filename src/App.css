#root {
	max-width: 1280px;
	margin: 0 auto;
	padding: 2rem;
	text-align: center;
}

.re-Counter {
	font-size: 3.5rem;
}

.Counter--positive {
	color: green;
}

.Counter--negative {
	color: red;
}

.re-Counter-wrapButtons {
	display: flex;
	justify-content: center;
	gap: 1rem;
}

@media screen and (max-width: 375px) {
	.re-Counter {
		font-size: 3rem;
	}

	.re-Counter-wrapButtons {
		flex-direction: column;
	}
}
