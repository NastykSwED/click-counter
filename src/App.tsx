import './App.css';
import { useCounter } from './hooks/useCounter';

const App = () => {
	const { counter, increment, decrement, reset } = useCounter(0);

	return (
		<div className='re-CounterApp'>
			<h1 className='re-CounterTitle'>Simple Click Counter</h1>
			<p
				className={`re-Counter ${
					counter > 0
						? 'Counter--positive'
						: counter < 0
						? 'Counter--negative'
						: ''
				}`}
			>
				{counter}
			</p>
			<div className='re-Counter-wrapButtons'>
				<button
					className='re-Counter-button Counter-button--increment'
					onClick={() => increment()}
				>
					Increment
				</button>
				<button
					className='re-Counter-button Counter-button--reset'
					onClick={() => reset()}
				>
					Reset
				</button>
				<button
					className='re-Counter-button Counter-button--decrement'
					onClick={() => decrement()}
				>
					Decrement
				</button>
			</div>
		</div>
	);
};

export default App;
