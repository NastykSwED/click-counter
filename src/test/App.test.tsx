import { describe, expect, it, afterEach } from 'vitest';

import { render, screen, fireEvent, cleanup } from '@testing-library/react';

import App from '../App';

describe('Testing Click Counter App', () => {
	afterEach(cleanup);

	it('Should match the snapshot', () => {
		const { container } = render(<App />);

		expect(container).toMatchSnapshot();
	});

	it('Should increase the counter value', () => {
		render(<App />);

		fireEvent.click(screen.getByText('Increment'));

		expect(screen.getByText(1).innerHTML).toBe('1');
	});

	it('Should decrease the counter value', () => {
		render(<App />);

		fireEvent.click(screen.getByText('Decrement'));

		expect(screen.getByText(-1).innerHTML).toBe('-1');
	});

	it('Should reset the counter value to zero', () => {
		render(<App />);

		fireEvent.click(screen.getByText('Increment'));

		expect(screen.getByText('1').innerHTML).toBe('1');

		fireEvent.click(screen.getByText('Reset'));

		expect(screen.getByText('0').innerHTML).toBe('0');
	});
});
