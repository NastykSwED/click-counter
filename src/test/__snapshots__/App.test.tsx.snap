// Vitest Snapshot v1

exports[`Testing Click Counter App > Should match the snapshot 1`] = `
<div>
  <div
    class="re-CounterApp"
  >
    <h1
      class="re-CounterTitle"
    >
      Simple Click Counter
    </h1>
    <p
      class="re-Counter"
    >
      0
    </p>
    <div
      class="re-Counter-wrapButtons"
    >
      <button
        class="re-Counter-button Counter-button--increment"
      >
        Increment
      </button>
      <button
        class="re-Counter-button Counter-button--reset"
      >
        Reset
      </button>
      <button
        class="re-Counter-button Counter-button--decrement"
      >
        Decrement
      </button>
    </div>
  </div>
</div>
`;
